platform/ios/src/MGLMapView.mm

```diff
@@ -2304,11 +2304,11 @@ - (void)installAnnotationImage:(MGLAnnotationImage *)annotationImage
     size_t width = CGImageGetWidth(image);
     size_t height = CGImageGetHeight(image);
     CGColorSpaceRef colorSpace = CGColorSpaceCreateDeviceRGB();
-    std::string pixels(width * height * 4, '\0');
+    mbgl::PremultipliedImage cPremultipliedImage(width, height);
     size_t bytesPerPixel = 4;
     size_t bytesPerRow = bytesPerPixel * width;
     size_t bitsPerComponent = 8;
-    char *pixelData = const_cast<char *>(pixels.data());
+    char *pixelData = reinterpret_cast<char *>(cPremultipliedImage.data.get());
     CGContextRef context = CGBitmapContextCreate(pixelData, width, height, bitsPerComponent, bytesPerRow, colorSpace, kCGImageAlphaPremultipliedLast);
```

Comment:
Avoid the cast and pass `cPremultipliedImage.data.get()` directly to `CGBitmapContextCreate`.